Userland/Libraries/LibGfx/Bitmap.cpp

```diff
@@ -583,15 +583,16 @@ Gfx::ShareableBitmap Bitmap::to_shareable_bitmap() const
     return Gfx::ShareableBitmap { bitmap_or_error.release_value_but_fixme_should_propagate_errors(), Gfx::ShareableBitmap::ConstructWithKnownGoodBitmap };
 }
 
-ErrorOr<BackingStore> Bitmap::allocate_backing_store(BitmapFormat format, IntSize size, int scale_factor)
+ErrorOr<BackingStore> Bitmap::allocate_backing_store(BitmapFormat format, IntSize size, int scale_factor, size_t pitch)
 {
     if (size.is_empty())
         return Error::from_string_literal("Gfx::Bitmap backing store size is empty");
 
     if (size_would_overflow(format, size, scale_factor))
         return Error::from_string_literal("Gfx::Bitmap backing store size overflow");
 
-    auto const pitch = minimum_pitch(size.width() * scale_factor, format);
+    if (pitch == 0)
+        pitch = minimum_pitch(size.width() * scale_factor, format);
```

Comment:
I think it would be more idiomatic to use an `Optional<size_t>` instead of 0 being a special value. But it's not a huge deal.

Reply:
i chose 0 because realistically pitch can never be smaller than width*bit_depth; but this is a good point, and I'll look into changing this :3